src/App.js

```diff
@@ -1,13 +1,69 @@
 import React from 'react';
+import { GoodsList } from './GoodsList';
 
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+class App extends React.Component {
+  state = {
+    goodList: [],
+  }
+
+  render() {
+    const { goodList } = this.state;
+
+    return (
+      <>
+        <h1>Dynamic list of Goods</h1>
+        <button
+          type="button"
+          onClick={() => {
+            getAll().then((goods) => {
+              this.setState({ goodList: goods });
+            });
+          }}
```

Comment:
First - such things is better to use with class method.
`render` should be as logic-free as possible.

Second - delegate all list preparations to api file.